### PR TITLE
Error handling

### LINKED ISSUE #10: Friendly exception reporting and errors messages
Don't through unnecessary stack traces for user errors. 
Show contextual, meaningful information that helps users locate and fix errors.

### PR DESCRIPTION
**Pull request by dcmoura**
Reporting of errors and warnings, providing context.
Closes #10.

e.g.
```
ERROR	could not evaluate SELECT expression #4: NULL if col1==1 else "hello" + 1
	at data row #2: [5, '5th line of test', -1, NULL]
TypeError: can only concatenate str (not "int") to str
```

### CODE REVIEW

File: spyql/processor.py [hunk 1]
```diff
@@ -109,7 +110,10 @@ def prepare_expression(self, expr):
 
         return [self.strings.put_strings_back(expr)]
 
-    def compile_clause(self, clause, clause_modifier = None, single = True, mode = 'eval'):
+    def is_clause_single(self, clause):
+        return clause not in ['select']
```
Comment (recharte) [inline spyql/processor.py:113]: :+1: Good job avoiding the boolean parameter in `compile_clause()` and `eval_clause()`

File: spyql/log.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+error_on_warning = False
+
+def user_error(message, exception, code = None, vars = None):
+    """
+    Reports an error, throwing the original exception
+    Prints a custom message.
+    Prints the data that originated the exception (if available).
+    """
+    sys.tracebacklimit = 0
+    logging.error(f"{colors.lightred}ERROR{colors.reset}\t{message}{': ' + colors.lightyellow + str(code) + colors.reset if code else ''}")
```
Comment (recharte) [inline spyql/log.py:15]: Does this conditional apply to the entire `': ' + colors.lightyellow + str(code) + colors.reset`?

**Review by recharte — APPROVED** [review]
Good job :+1:

File: spyql/log.py (hunk 2 above)
Comment (dcmoura) [reply spyql/log.py:15]: Yes. Basically returns an empty string if code is not present, otherwise adds ':' followed by ' 'yellow' code.

**Pull request merged**